### ISSUE TITLE
unsuccessful make install

### ISSUE THREAD

**Issue by pedramardakani** [Perl/perl5#17712]
Module:

**Description**

Can not produce a successful build from source. To give you more specific description I ran the `make test` command. As you can see some parts are skipped or failed:
```
t/run/dtrace ................................................... skipped
t/run/switchDx ................................................. skipped
t/io/eintr ..................................................... skipped
t/io/eintr_print ............................................... skipped
t/io/fs ........................................................ FAILED at test 9
t/re/fold_grind_T .............................................. skipped
t/op/dump ...................................................... skipped
t/op/filetest .................................................. FAILED at test 6
t/op/require_errors ............................................ FAILED--unexpected output at test 36
t/op/stat ...................................................... FAILED at test 19
t/uni/cache .................................................... skipped
t/uni/greek .................................................... skipped
t/uni/latin2 ................................................... skipped
t/lib/cygwin ................................................... skipped
lib/AnyDBM_File ................................................ FAILED at test 2
lib/File/Copy .................................................. FAILED at test 73
lib/Net/netent ................................................. skipped
lib/perl5db .................................................... FAILED at test 1
ext/File-Glob/t/basic .......................................... FAILED at test 9
ext/Pod-Html/t/cache ........................................... FAILED at test 8
ext/POSIX/t/termios ............................................ FAILED at test 98
ext/SDBM_File/t/sdbm ........................................... FAILED at test 4
ext/XS-APItest/t/win32 ......................................... skipped
dist/base/t/fields-5_6_0 ....................................... skipped
dist/base/t/fields-5_8_0 ....................................... skipped
dist/Data-Dumper/t/huge ........................................ skipped
dist/encoding-warnings/t/2-fatal ............................... skipped
dist/encoding-warnings/t/3-normal .............................. skipped
dist/encoding-warnings/t/4-lexical ............................. skipped
dist/Module-CoreList/t/pod ..................................... skipped
dist/Net-Ping/t/190_alarm ...................................... skipped
dist/Net-Ping/t/200_ping_tcp ................................... skipped
dist/Net-Ping/t/250_ping_hires ................................. skipped
dist/Net-Ping/t/300_ping_stream ................................ skipped
dist/Net-Ping/t/400_ping_syn ................................... skipped
dist/Net-Ping/t/410_syn_host ................................... skipped
dist/Net-Ping/t/420_ping_syn_port .............................. skipped
dist/PathTools/t/win32 ......................................... skipped
dist/Storable/t/compat01 ....................................... skipped
dist/Storable/t/huge ........................................... skipped
dist/Storable/t/hugeids ........................................ skipped
dist/Storable/t/interwork56 .................................... skipped
dist/Storable/t/leaks .......................................... skipped
dist/Term-ReadLine/t/AE ........................................ skipped
dist/Term-ReadLine/t/AETk ...................................... skipped
dist/Term-ReadLine/t/Tk ........................................ skipped
dist/threads/t/no_threads ...................................... skipped
dist/threads/t/unique .......................................... skipped
dist/Tie-File/t/21_win32 ....................................... skipped
dist/Time-HiRes/t/utime ........................................ FAILED at test 2
cpan/Archive-Tar/t/04_resolved_issues .......................... FAILED at test 56
cpan/Archive-Tar/t/90_symlink .................................. skipped
cpan/Archive-Tar/t/99_pod ...................................... skipped
cpan/autodie/t/import-into ..................................... skipped
cpan/autodie/t/version ......................................... skipped
cpan/bignum/t/bn_lite .......................................... skipped
cpan/bignum/t/br_lite .......................................... skipped
cpan/Compress-Raw-Bzip2/t/99pod ................................ skipped
cpan/Compress-Raw-Zlib/t/07bufsize ............................. skipped
cpan/Encode/t/enc_data ......................................... skipped
cpan/Encode/t/enc_eucjp ........................................ skipped
cpan/Encode/t/enc_module ....................................... skipped
cpan/Encode/t/enc_utf8 ......................................... skipped
cpan/Encode/t/encoding ......................................... skipped
cpan/Encode/t/jperl ............................................ skipped
cpan/Encode/t/piconv ........................................... skipped
cpan/Encode/t/whatwg-aliases ................................... skipped
cpan/ExtUtils-Install/t/Install ................................ FAILED at test 52
cpan/ExtUtils-Install/t/InstallWithMM .......................... FAILED at test 4
cpan/ExtUtils-MakeMaker/t/00compile ............................ skipped
cpan/ExtUtils-MakeMaker/t/03-xsstatic .......................... skipped
cpan/ExtUtils-MakeMaker/t/basic ................................ FAILED at test 34
cpan/ExtUtils-MakeMaker/t/eu_command ........................... FAILED at test 11
cpan/ExtUtils-MakeMaker/t/INSTALL_BASE ......................... FAILED at test 5
cpan/ExtUtils-MakeMaker/t/miniperl ............................. FAILED at test 4
cpan/ExtUtils-MakeMaker/t/MM_BeOS .............................. skipped
cpan/ExtUtils-MakeMaker/t/MM_Cygwin ............................ skipped
cpan/ExtUtils-MakeMaker/t/MM_NW5 ............................... skipped
cpan/ExtUtils-MakeMaker/t/MM_OS2 ............................... skipped
cpan/ExtUtils-MakeMaker/t/MM_Unix .............................. FAILED at test 96
cpan/ExtUtils-MakeMaker/t/MM_VMS ............................... skipped
cpan/ExtUtils-MakeMaker/t/MM_Win32 ............................. skipped
cpan/ExtUtils-MakeMaker/t/PL_FILES ............................. FAILED at test 3
cpan/ExtUtils-MakeMaker/t/pm_to_blib ........................... FAILED at test 4
cpan/ExtUtils-MakeMaker/t/unicode .............................. FAILED at test 4
cpan/File-Path/t/Path .......................................... FAILED--expected 165 tests, saw 62
cpan/File-Path/t/Path_root ..................................... skipped
cpan/File-Path/t/Path_win32 .................................... skipped
cpan/File-Temp/t/lock .......................................... skipped
cpan/IO-Compress/t/020isize .................................... skipped
cpan/IO-Compress/t/101truncate-bzip2 ........................... skipped
cpan/IO-Compress/t/101truncate-deflate ......................... skipped
cpan/IO-Compress/t/101truncate-gzip ............................ skipped
cpan/IO-Compress/t/101truncate-rawdeflate ...................... skipped
cpan/IO-Compress/t/101truncate-zip ............................. skipped
cpan/IO-Compress/t/105oneshot-zip-store-only ................... skipped
cpan/IO-Compress/t/106prime-bzip2 .............................. skipped
cpan/IO-Compress/t/106prime-deflate ............................ skipped
cpan/IO-Compress/t/106prime-gzip ............................... skipped
cpan/IO-Compress/t/106prime-rawdeflate ......................... skipped
cpan/IO-Compress/t/106prime-zip ................................ skipped
cpan/IO-Compress/t/109merge-zip ................................ skipped
cpan/IO-Compress/t/999pod ...................................... skipped
cpan/IO-Socket-IP/t/99pod ...................................... skipped
cpan/IPC-Cmd/t/02_Interactive .................................. skipped
cpan/IPC-SysV/t/pod ............................................ skipped
cpan/IPC-SysV/t/podcov ......................................... skipped
cpan/libnet/t/ftp .............................................. skipped
cpan/libnet/t/nntp ............................................. skipped
cpan/libnet/t/nntp_ssl ......................................... skipped
cpan/libnet/t/pop3_ssl ......................................... skipped
cpan/libnet/t/smtp ............................................. skipped
cpan/libnet/t/smtp_ssl ......................................... skipped
cpan/Locale-Maketext-Simple/t/2-load_po_without_i_default ...... skipped
cpan/Locale-Maketext-Simple/t/3-load_po_with_i_default ......... skipped
cpan/Memoize/t/tie_gdbm ........................................ skipped
cpan/Memoize/t/tie_ndbm ........................................ skipped
cpan/Pod-Perldoc/t/pod ......................................... skipped
cpan/podlators/t/docs/pod ...................................... skipped
cpan/podlators/t/docs/pod-spelling ............................. skipped
cpan/podlators/t/docs/spdx-license ............................. skipped
cpan/podlators/t/docs/synopsis ................................. skipped
cpan/podlators/t/docs/urls ..................................... skipped
cpan/podlators/t/style/minimum-version ......................... skipped
cpan/podlators/t/style/module-version .......................... skipped
cpan/podlators/t/style/strict .................................. skipped
cpan/Sys-Syslog/t/facilities-routing ........................... skipped
cpan/Test-Harness/t/compat/nonumbers ........................... skipped
cpan/Test-Simple/t/Legacy/is_deeply_with_threads ............... skipped
cpan/Test-Simple/t/Legacy/plan_skip_all ........................ skipped
cpan/Test-Simple/t/Legacy/Regression/637 ....................... skipped
cpan/Test-Simple/t/Legacy/Regression/789-read-only ............. skipped
cpan/Test-Simple/t/Legacy/strays ............................... skipped
cpan/Test-Simple/t/Test2/acceptance/try_it_skip ................ skipped
cpan/Test-Simple/t/Test2/regression/gh_16 ...................... skipped
cpan/Test-Simple/t/Test2/regression/ipc_files_abort_exit ....... skipped
cpan/version/t/11_taint ........................................ skipped
t/porting/authors .............................................. skipped
t/porting/bench ................................................ skipped
t/porting/cmp_version .......................................... skipped
t/porting/exec-bit ............................................. FAILED at test 1
t/porting/libperl .............................................. skipped
t/porting/pending-author ....................................... skipped
t/porting/utils ................................................ skipped
```
**Steps to Reproduce**

Run `make install` after depackaging the *perl-5.30.0.tar.gz* file.

**Expected behavior**

Successful build from source after executing `make install`.

**Perl configuration**

```
Summary of my perl5 (revision 5 version 30 subversion 0) configuration:
   
  Platform:
    osname=linux
    osvers=5.0.0-38-generic
    archname=x86_64-linux-thread-multi
    uname='linux pedram-pc 5.0.0-38-generic #41-ubuntu smp tue dec 3 00:27:35 utc 2019 x86_64 x86_64 x86_64 gnulinux '
    config_args='-des -Dusethreads -Duseshrplib -Dprefix=/build/dir -Dvendorprefix=/build/dir -Dman1ext=1perl -Dman3ext=3perl -Dcccdlflags=-fPIC'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='cc'
    ccflags ='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    optimize='-O2'
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='8.3.0'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib/gcc/x86_64-linux-gnu/8/include-fixed /usr/include/x86_64-linux-gnu /usr/lib /lib/x86_64-linux-gnu /lib/../lib /usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib /lib64 /usr/lib64
    libs=-lpthread -lnsl -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -lnsl -ldl -lm -lcrypt -lutil -lc
    libc=libc-2.29.so
    so=so
    useshrplib=true
    libperl=libperl.so
    gnulibc_version='2.29'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E -Wl,-rpath,/build/dir/lib/5.30.0/x86_64-linux-thread-multi/CORE'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_TIMES
    MULTIPLICITY
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_IMPLICIT_CONTEXT
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_ITHREADS
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
    USE_REENTRANT_API
    USE_THREAD_SAFE_LOCALE
  Built under linux
  Compiled at Apr  9 2020 10:09:46
  @INC:
    /build/dir/lib/site_perl/5.30.0/x86_64-linux-thread-multi
    /build/dir/lib/site_perl/5.30.0
    /build/dir/lib/vendor_perl/5.30.0/x86_64-linux-thread-multi
    /build/dir/lib/vendor_perl/5.30.0
    /build/dir/lib/5.30.0/x86_64-linux-thread-multi
    /build/dir/lib/5.30.0

```

**Comment by jkeenan** [comment]
> 
> Can not produce a successful build from source. To give you more specific description I ran the `make test` command. As you can see some parts are skipped or failed:
> 
> ```
> t/run/dtrace ................................................... skipped
> t/run/switchDx ................................................. skipped
> ```
...

Yes, that is an unusual pattern of skipped tests.  Let's start from the beginning.  (Note:  Since this is best approached interactively, if you have IRC access, please go to irc.perl.org  \#p5p 

1. Go to the directory into which you unwrapped your tarball and call:
```
$ make realclean; rm -v config.sh Policy.sh
```
This should get that directory back to its original, unwrapped state.  (Alternatively, just wipe out that directory and unwrap the tarball again.)

2. Call:  `$ sh ./Configure -des -Dusedevel`.  This probes your machine to get configuration data and write the `makefile`.

3. Call:  `make test_prep`.  This builds the new `perl` executable and all other files needed to run the test suite.

4. At this point, you should be able to call either `make test` or `make test_harness` to run the test suite.  However, since that can be time-consuming, you may wish to run a small subset of the tests that were failed or skipped on your first run.  You would do that with an invocation like this:
```
$ cd t; ./perl harness -v io/*.t; cd -
```
Then report back what you observe.

Thank you very much.
Jim Keenan

**Comment by pedramardakani** [comment]
Thank you for your response, Jim.

1. I chose to do the alternative that you suggested (`rm -r` directory and extracted the tarball again)
2. I called: `$ sh ./Configure -des -Dusedevel | tee bug-config.txt` to save the output for your info.
3. Again as you mentioned: `$ make test_prep | tee bug-test-prep.txt`, used `tee` for the same reason.
4. Then I ran the `$ make test_harness | tee bug-test-harness.txt` to get as much info as I could.
5. As a result of the last step, I got the error:
```
make: *** [makefile:837: test_harness] Error 114
```
All the log files are available in the file attached here. I do appreciate your help.

Cheers,
Pedram

[logs.tar.gz](https://github.com/Perl/perl5/files/4465247/logs.tar.gz)

**Comment by jkeenan** [comment]
> Thank you for your response, Jim.
> 
>     1. I chose to do the alternative that you suggested (`rm -r` directory and extracted the tarball again)
> 
>     2. I called: `$ sh ./Configure -des -Dusedevel | tee bug-config.txt` to save the output for your info.
> 
>     3. Again as you mentioned: `$ make test_prep | tee bug-test-prep.txt`, used `tee` for the same reason.
> 
>     4. Then I ran the `$ make test_harness | tee bug-test-harness.txt` to get as much info as I could.
> 
>     5. As a result of the last step, I got the error:
> 
> 
> ```
> make: *** [makefile:837: test_harness] Error 114
> ```
> 
> All the log files are available in the file attached here. I do appreciate your help.
> 
> Cheers,
> Pedram
> 
> [logs.tar.gz](https://github.com/Perl/perl5/files/4465247/logs.tar.gz)

Yes, that is a very startling number of test-time errors.  Your configuration and build output look fine.

So I suspect something is wrong with your system.  Let's try running some of the more fundamental test files.  Run either these:
```
$> ./perl -Ilib t/io/fs.t
$> ./perl -Ilib t/op/filetest.t
$> ./perl -Ilib t/op/require_errors.t
$> ./perl -Ilib t/op/stat.t
```
or, what amounts to the same thing, these:
```
$> cd t;./perl harness -v io/fs.t; cd -
$> cd t;./perl harness -v op/filetest.t; cd -
$> cd t;./perl harness -v op/require_errors.t; cd -
$> cd t;./perl harness -v op/stat.t; cd -
```

**Comment by jkeenan** [comment]
The test failures appear to be associated with file permissions problems and `chmod`.  What you should be getting looks like this:
```
$ ./perl -Ilib t/io/fs.t
1..61
ok 1 - umask
ok 2 - link a b
ok 3 - link b c
ok 4 - link count of triply-linked file
ok 5 - mode of triply-linked file
ok 6 - chmod succeeding
ok 7 - chmod going through
ok 8 - chmod two files
ok 9 - chmod going through to c
ok 10 - chmod going through to x
ok 11 - unlink two files
ok 12 - ino of removed file b should be undef
ok 13 - ino of removed file x should be undef
ok 14 - open a
ok 15 - fchmod
ok 16 - perm reset
ok 17 - fchmod
ok 18 - perm restored
...

$ ./perl -Ilib t/op/filetest.t
1..435
ok 1 - -d: directory correctly identified
ok 2 - -f: plain file correctly identified
ok 3 - -f: directory is not a plain file
ok 4 - -d: plain file is not a directory
ok 5 - -r: file readable by effective uid/gid not found
ok 6 - -w: file writable by effective uid/gid
...

$ ./perl -Ilib t/op/filetest.t
ok 18 - chmod 0000
ok 19 -    -r
ok 20 -    -w
ok 21 -    -x
...
ok 90 - -T _ doesn't break the statbuffer
ok 91 - -T _ doesn't break lstat for unreadable file
...
```
Do you have some sort of mask operating there?

**Comment by pedramardakani** [comment]
I changed directory to `$ cd /unpacked/perl/tarball/dir/t`, ran all the four commands, and saved the outputs to corresponding files. As a bonus, I called `$ ./perl -p harness -v op/require_errors.t | tee bug-require-errors-p.txt` to get the best details I could. The outputs are compressed in a GZ file attached in this comment.
```
$ ./perl harness -v io/fs.t | tee bug-io-fs.txt
$ ./perl harness -v op/filetest.t | tee bug-filetest.txt
$ ./perl harness -v op/require_errors.t | tee bug-require-errors.txt
$ ./perl -p harness -v op/require_errors.t | tee bug-require-errors-p.txt
$ ./perl harness -v op/stat.t | tee bug-stat.txt
$ tar cvaf bug-harness-quad.gz *.txt
```
But I don't know what should I do. Should I change the `chmod` of any files manually or something? Completely run the make on `sudo`? (If that is the case, it breaks the whole point for me, as I'm trying to add this Perl part to some other project that should not ask for root privileges)

About the mask operations, I really don't know! I'd very much appreciate it if you'd enlighten me on that matter. All I'm doing is just unpacking the tarball and trying to install the Perl5.30. I'm trying my best to provide you with enough information. Please let me know if I can help further. 

BTW, I have never used the irc.perl.org. If you think that helps better, please guide me through it and I'll do the best I can. Thanks in advance!

[bug-harness-quad.gz](https://github.com/Perl/perl5/files/4465930/bug-harness-quad.gz) (~8 MB)

**Comment by Grinnz** [comment]
You can use a web client like [kiwiirc](https://kiwiirc.com/nextclient/#irc://irc.perl.org/#p5p) to join the IRC channel.

**Comment by pedramardakani** [comment]
Thanks guys (@jkeenan @Grinnz and *xenu* on *irc.perl.org/#p5p*) ! So, it looks like Perl shouldn't be built on **NTFS** partitions. *xenu* noted that and believed that **NTFS** couldn't handle UNIX file permissions properly. I simply changed the build directory and the build problem was solved.

**Comment by pedramardakani** [comment]
Shouldn't Perl have the ability to check and warn the user that building in **NTFS** wouldn't work?

**Comment by jkeenan** [comment]
> NTFS

You should have mentioned NTFS in your original post.  That would have saved a lot of time and effort, as people familiar with NTFS (which I am not) would have immediately honed in on that aspect of the problem.  Everything in your original post suggested that you were working on Linux.

**Comment by pedramardakani** [comment]
@jkeenan It wouldn't be an issue if I knew that :-) That's how we learn. I hope you do understand that I'm not here to waste **your** time and effort. Me and my friends have put so much time into this too. This is the last place I visited, and your mentioning of the *irc* channel did help a lot. Asking about the media I ran the build on was the first question *xenu* asked. So I guess she/he has faced some similar problem somewhere. Anyways, if this is **SO** obvious, the program **should** complain about it **before** "wasting so much time and effort".

I hope this thread helps someone somewhere. Or put an **idea** into a developer's mind to prevent this problem with an **early test** on the media type the program is being built on. Since that is the purpose here. 

Cheers.
